Assets/Scripts/StateMachine/StateMachine.cs

```diff
@@ -79,5 +87,16 @@ public State GetState<TState>() where TState : State, new()
             State parameterState = new TState();
             return States.FirstOrDefault(state => state.ToString() == parameterState.ToString()) ?? parameterState;
         }
+
+        /// <summary>
+        /// Returns the reference of a state based on the monoscript param in the StateMachine. Will create new instance if not yet available.
+        /// </summary>
+        public State GetState(MonoScript monoScript)
+        {
+            return States.FirstOrDefault
+            (
+                s => monoScript.GetClass().ToString().Contains(s.ToString())
+            ) ?? (State)monoScript.GetClass().GetConstructor(new System.Type[] { }).Invoke(new object[] { });
+        }
```

Comment:
```suggestion
public State GetState(MonoScript monoScript)
{
    return States.FirstOrDefault(s => monoScript.GetClass().ToString().Contains(s.ToString())) 
           ?? (State)monoScript.GetClass().GetConstructor(new Type[] { }).Invoke(new object[] { });
}
```
looks a little less cursed